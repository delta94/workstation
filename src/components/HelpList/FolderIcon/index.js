import React from 'react';
import cc from 'classcat';

import { ReactComponent as Category } from './icons/Category.svg';
import { ReactComponent as CategoryActive } from './icons/CategoryActive.svg';

import classes from './styles.module.scss';

export default function FolderIcon({ isActive, className = '' }) {
  const IconComponent = isActive ? CategoryActive : Category;
  return <IconComponent className={cc([classes['folder-icon'], className])} />;
}
