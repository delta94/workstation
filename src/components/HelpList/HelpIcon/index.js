import React from 'react';
import cc from 'classcat';

import { ReactComponent as Article } from './icons/Article.svg';
import { ReactComponent as Shuttle } from './icons/Shuttle.svg';
import { ReactComponent as Survey } from './icons/Survey.svg';
import { ReactComponent as Video } from './icons/Video.svg';
import { ReactComponent as Walkthru } from './icons/Walkthru.svg';

import classes from './styles.module.scss';

const iconsByType = {
  article: Article,
  shuttle: Shuttle,
  video: Video,
  'smart-walkthru': Walkthru,
  survey: Survey,
};

export default function HelpIcon({ type, className = '' }) {
  const IconComponent = iconsByType[type];
  return <IconComponent className={cc([classes['help-icon'], className])} />;
}
