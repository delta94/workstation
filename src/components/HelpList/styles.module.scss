.help-list {
  --icon-width: 34px;
  list-style-type: none;
  margin: 30px 0;
  padding: 0;

  .help-group {
    border-top: solid 1px var(--border100);

    &.last-of-type {
      border-bottom: solid 1px var(--border100);
    }

    .help-group-title {
      font-size: 16px;
      font-family: var(--font-primary);
      font-weight: 500;
      padding: 20px 30px;
      cursor: pointer;
      color: var(--neutral900);

      &:hover {
        color: var(--primary);
        background-color: var(--secondary100);

        .help-list-icon {
          color: var(--primary);
        }
      }

      .help-list-icon {
        display: inline-block;
        margin-right: 17px;
        color: var(--secondary500);
      }
    }

    ul {
      padding-left: 0;

      li {
        padding-left: 50px;
      }
    }
  }
}

.help-item {
  width: 100%;
  margin-left: 0;
  padding: 17px 30px;
  cursor: pointer;
  display: flex;
  flex-direction: row;

  &:hover {
    .help-list-icon,
    .title {
      color: var(--primary);
    }
  }

  .help-list-icon {
    display: inline-block;
    width: var(--icon-width);
    color: var(--secondary500);
    flex-shrink: 0;
  }

  .help-item-text {
    width: calc(100% - var(--icon-width));
  }

  .title {
    color: var(--neutral900);
    font-weight: 500;
    font-size: 15px;
    vertical-align: bottom;
  }

  .description {
    font-size: 13px;
    line-height: 1.6;
    color: var(--neutral500);
    padding-top: 4px;
  }

  .url {
    color: var(--secondary500);
    font-weight: 100;
    font-size: 13px;

    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }

  &:hover {
    background-color: var(--secondary100);
  }
}
