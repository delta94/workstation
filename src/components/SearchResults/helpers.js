import walkme from '@walkme/sdk';
import uniqWith from 'lodash/uniqWith';
import isEqual from 'lodash/isEqual';

function findInNode(node, searchTerm) {
  const description = node.description ?? '';
  const url = node.properties.url ?? '';
  const keywords = node.keywords?.join(' ') ?? '';

  return [node.title, description, url, keywords].some((string) =>
    string.toLowerCase().includes(searchTerm.toLowerCase())
  );
}

function flattenUITree(uiTree) {
  const flattenedContent = [];

  uiTree
    .map((tab) => tab.childNodes)
    .flat(1)
    .forEach((node) => {
      node.type === 'category' ? flattenedContent.push(...node.childNodes) : flattenedContent.push(node);
    });

  return uniqWith(flattenedContent, isEqual);
}

export function findInUiTree(searchTerm, uiTree) {
  return (
    flattenUITree(uiTree)
      .filter((node) => ![walkme.content.TypeNames.Survey].includes(node.type))
      // get items that include the search term in their attributes
      .filter((node) => findInNode(node, searchTerm))
  );
}

export async function findInSearchApi(searchTerm, wmSearch) {
  try {
    const searchResults = await wmSearch.search(searchTerm);
    return searchResults.filter((item) => item.type === walkme.content.TypeNames.SearchResult);
  } catch (err) {
    console.error('Could not get search results from the API.', err);
  }
}
