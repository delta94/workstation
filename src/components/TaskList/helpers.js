function flattenNodes(content) {
  const flattenedContent = [];

  content.forEach((node) => {
    node.type === 'category' ? flattenedContent.push(...node.childNodes) : flattenedContent.push(node);
  });

  return flattenedContent;
}

export function calculateCompletedPercent(content) {
  const flatTasks = flattenNodes(content);
  const total = flatTasks.length;
  const completed = flatTasks.filter((node) => node.properties.isCompleted).length;
  return Math.round((completed / total) * 100);
}
