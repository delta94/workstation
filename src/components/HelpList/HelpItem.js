import React from 'react';
import walkme from '@walkme/sdk';

import HelpIcon from './HelpIcon';

import classes from './styles.module.scss';

export default function HelpItem({ node }) {
  function onClick() {
    walkme.content.playById(node.type, node.id);
    walkme.platform.closeMe();
  }

  return (
    <li className={classes['help-item']} key={node.id} onClick={onClick}>
      <div className={classes['help-list-icon']}>
        <HelpIcon type={node.type} />
      </div>
      <div className={classes['help-item-text']}>
        <div className={classes.title}>{node.title}</div>
        <div className={classes.description}>{node.description}</div>
        {node.properties?.url && <div className={classes.url}>{node.properties?.url}</div>}
      </div>
    </li>
  );
}
