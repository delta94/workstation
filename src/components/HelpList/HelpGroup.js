import React, { useState } from 'react';
import cc from 'classcat';
import SmoothCollapse from 'react-smooth-collapse';

import HelpItem from './HelpItem';

import FolderIcon from './FolderIcon';
import classes from './styles.module.scss';

export default function HelpList({ node, className = '' }) {
  const [isOpen, setIsOpen] = useState(false);

  function toggleDropDown() {
    setIsOpen(!isOpen);
  }

  return (
    <li className={cc([classes['help-group'], className])}>
      <div className={classes['help-group-title']} onClick={toggleDropDown}>
        <FolderIcon isActive={isOpen} className={classes['help-list-icon']} />
        <span>{node.title}</span>
      </div>
      <SmoothCollapse expanded={isOpen}>
        <ul>
          {node.childNodes?.map((childNode) => (
            <HelpItem key={childNode.id} node={childNode} />
          ))}
        </ul>
      </SmoothCollapse>
    </li>
  );
}
