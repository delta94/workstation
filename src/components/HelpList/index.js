import React from 'react';
import cc from 'classcat';

import HelpItem from './HelpItem';
import HelpGroup from './HelpGroup';

import classes from './styles.module.scss';

export default function HelpList({ content }) {
  return (
    <ul className={classes['help-list']}>
      {content.map((node, index) => {
        if (node.type === 'category') {
          const isLastOfType = Boolean(index === content.length - 1 || content[index + 1].type !== 'category');
          return <HelpGroup key={node.id} node={node} className={cc([{ [classes['last-of-type']]: isLastOfType }])} />;
        }

        return <HelpItem key={node.id} node={node} />;
      })}
    </ul>
  );
}
