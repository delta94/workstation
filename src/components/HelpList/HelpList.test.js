import React from 'react';
import { render } from '@testing-library/react';

import { sdkProvider } from '../../helpers/sdkMockForTests';
import HelpList from './index';

describe('HelpList renders correctly', () => {
  test('should render correctly', () => {
    const { container } = render(<HelpList content={sdkProvider.state.sdk.uiTreeSDK[0].childNodes} />);
    expect(container).toMatchSnapshot();
  });
});
