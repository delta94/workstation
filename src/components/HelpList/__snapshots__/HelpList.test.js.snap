// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpList renders correctly should render correctly 1`] = `
<div>
  <ul
    class="help-list"
  >
    <li
      class="help-item"
    >
      <div
        class="help-list-icon"
      >
        <svg
          class="help-icon"
        >
          Walkthru.svg
        </svg>
      </div>
      <div
        class="help-item-text"
      >
        <div
          class="title"
        >
          Content smart-walkthru-280
        </div>
        <div
          class="description"
        >
          Description for smart-walkthru-280
        </div>
      </div>
    </li>
    <li
      class="help-item"
    >
      <div
        class="help-list-icon"
      >
        <svg
          class="help-icon"
        >
          Survey.svg
        </svg>
      </div>
      <div
        class="help-item-text"
      >
        <div
          class="title"
        >
          Content survey-777
        </div>
        <div
          class="description"
        >
          Description for survey-777
        </div>
      </div>
    </li>
    <li
      class="help-item"
    >
      <div
        class="help-list-icon"
      >
        <svg
          class="help-icon"
        >
          Shuttle.svg
        </svg>
      </div>
      <div
        class="help-item-text"
      >
        <div
          class="title"
        >
          Content shuttle-27
        </div>
        <div
          class="description"
        >
          Description for shuttle-27
        </div>
        <div
          class="url"
        >
          https://27.example.com/index?t=599
        </div>
      </div>
    </li>
    <li
      class="help-item"
    >
      <div
        class="help-list-icon"
      >
        <svg
          class="help-icon"
        >
          Video.svg
        </svg>
      </div>
      <div
        class="help-item-text"
      >
        <div
          class="title"
        >
          Content video-981
        </div>
        <div
          class="description"
        >
          Description for video-981
        </div>
      </div>
    </li>
    <li
      class="help-item"
    >
      <div
        class="help-list-icon"
      >
        <svg
          class="help-icon"
        >
          Article.svg
        </svg>
      </div>
      <div
        class="help-item-text"
      >
        <div
          class="title"
        >
          Content article-723
        </div>
        <div
          class="description"
        >
          Description for article-723
        </div>
      </div>
    </li>
    <li
      class="help-group last-of-type"
    >
      <div
        class="help-group-title"
      >
        <svg
          class="folder-icon help-list-icon"
        >
          Category.svg
        </svg>
        <span>
          Category
        </span>
      </div>
      <div
        class=""
        style="height: 0px; overflow: hidden; display: none; transition: height .25s ease;"
      />
    </li>
  </ul>
</div>
`;
